client/organization.go

```diff
@@ -41,16 +44,39 @@ func (client *ApiClient) Organization() (Organization, error) {
 	return result[0], nil
 }
 
-func (client *ApiClient) OrganizationId() (string, error) {
-	if client.cachedOrganizationId != "" {
-		return client.cachedOrganizationId, nil
+func (client *ApiClient) getCachedOrganization() (*Organization, error) {
+	cachedOrganizationLock.Lock()
```

Comment:
Added a mutex. I want to prevent race conditions... (shouldn't occur. But just to be on the safe side).